### ISSUE TITLE
L2 loss in tensorboard keeps growing, what is the reason？

### ISSUE THREAD

**Issue by FancyFai** [google/automl#367]
![image](https://user-images.githubusercontent.com/48304999/81244321-b1008a80-9044-11ea-991f-b103dcd8dee9.png)
Does anyone have this problem? L2 loss is going up .Could you give me some Suggestions? Thank you very much
this is my command：
python main.py --mode=train_and_eval \
    --num_classes=6 \
    --training_file_pattern=tfrecord/train*.tfrecord \
    --validation_file_pattern=tfrecord/val*.tfrecord \
    --val_json_file=tfrecord/json_val.json \
    --model_name=efficientdet-d2 \
    --model_dir=tmp/efficientdet-d2-scratch_2Data_200  \
    --ckpt=efficientdet-ckpt/efficientdet-d2  \
    --train_batch_size=4 \
    --eval_batch_size=4 --eval_samples=1493 \
    --num_examples_per_epoch=11950 --num_epochs=200  \
    --hparams="num_classes=6,moving_average_decay=0" \
    --use_tpu=False
this is my ap results：
![image](https://user-images.githubusercontent.com/48304999/81244534-51ef4580-9045-11ea-89de-bf62a5f43fee.png)

**Comment by qtw1998** [comment]
Yeah that' the truth. Always like this.

**Comment by FancyFai** [comment]
> Yeah that' the truth. Always like this.

Oh, oh, but I remember that L2 loss is the square of the difference between the true value and the predicted value

**Comment by fsx950223** [comment]
try adam optimizer

**Comment by mingxingtan** [comment]
Interesting!  @FancyFai , l2 loss is the weight decay loss, not the cross-entropy loss.

I checked my trarining for COCO, and the l2 loss is dropping:

![image](https://user-images.githubusercontent.com/6027221/81786729-e8dc5780-94b4-11ea-90f7-5421944fac88.png)

If you are interested in the full log, you can download here: https://storage.googleapis.com/cloud-tpu-checkpoints/efficientdet/coco/tb/efficientdet-d0-tb.tgz

**Comment by mingxingtan** [comment]
L2 regularization loss is defined like this: https://www.tensorflow.org/api_docs/python/tf/nn/l2_loss

**Comment by fsx950223** [comment]
> L2 regularization loss is defined like this: https://www.tensorflow.org/api_docs/python/tf/nn/l2_loss

It's l2 regularization instead of LSE

**Comment by fitoule** [comment]
Same here L2 Loss is growing. I tested with adam optimizer following @fsx950223  advice but it is worst

**Comment by LinHungShi** [comment]
@fsx950223  I have the same problem. Have you solved it yet ? I am also wondering why the reg term becomes so small since weight decay is 4e-5 and the gradient for reg term becomes even smaller after multiplied by the learning rate.

**Comment by fsx950223** [comment]
> @fsx950223 I have the same problem. Have you solved it yet ? I am also wondering why the reg term becomes so small since weight decay is 4e-5 by default and the gradient for reg term becomes even smaller after multiplied by the learning rate.

It's normal when you try to train efficientdet ckpt with fewer classes.
